src/NNUnit.cpp

```diff
@@ -8,16 +8,10 @@ void NNUnit::computeValues(){
     if (_feeder){
         // unit value
         _pv = _feeder->getFeed();
-        _v = _actf->f(_pv);
         // shared useful values
-        double a1d = 0.;
-        if (_v1d || _v2d || _v1vd || _v1d1vd) a1d = _actf->f1d(_pv);
-
-        double a2d = 0.;
-        if (_v2d || _v1vd) a2d = _actf->f2d(_pv);
+        double a1d, a2d, a3d;
 
-        double a3d = 0.;
-        if (_v2d1vd) a3d = _actf->f3d(_pv);
+        _actf->fad(_pv, _v, a1d, a2d, a3d, _v1d || _v2d || _v1vd || _v1d1vd, _v2d || _v1vd, _v2d1vd);
```

Comment:
mmm... I am wondering if it is possible to improve readability... right now it is quite hard to read.
What about first defining some const bool with the flag values and then call the fad function?

```
const bool flag_actf_d1 = _v1d || _v2d || _v1vd || _v1d1vd;
const bool flag_actf_d2 = _v2d || _v1vd;
const bool flag_actf_d3 = _v2d1vd;
_actf->fad(_pv, _v, a1d, a2d, a3d, flag_actf_d1, flag_actf_d2, flag_actf_d3);
```